kroxylicious/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -115,6 +113,22 @@ else if (requestFilterResult.message() != null) {
         }
     }
 
+    private void forwardShortCircuitResponse(ChannelHandlerContext ctx, DecodedRequestFrame<?> decodedFrame, DefaultFilterContext filterContext,
+                                             RequestFilterResult requestFilterResult) {
+        if (decodedFrame.hasResponse()) {
+            var header = requestFilterResult.header() == null ? new ResponseHeaderData() : ((ResponseHeaderData) requestFilterResult.header());
+            header.setCorrelationId(decodedFrame.correlationId());
+            filterContext.forwardResponseInternal(header, requestFilterResult.message());
+        }
+        else {
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
Happy with how this looks for now for this PR.

I'm wondering if we should actually be failing early (in the builder) and detecting this case and the case where you try to short circuit with mismatched response.